source/utilities.tex

```diff
@@ -6463,6 +6464,2043 @@
+\item
+Otherwise, if \tcode{rhs.has_value()} is \tcode{true},
+destroys \exposid{unex} and sets \exposid{has_val} to \tcode{true}.
+\item
+Otherwise, equivalent to \tcode{\exposid{unex} = rhs.error()}.
+\end{itemize}
+
+\pnum
+\returns
+\tcode{*this}.
+
+\pnum
+\remarks
+The exception specification is
+\tcode{is_nothrow_move_constructible_v<E> \&\& is_nothrow_move_assignable_v<E>}.
+
+\pnum
+This operator is defined as deleted unless
+\tcode{is_move_constructible_v<E>} is \tcode{true} and
+\tcode{is_move_assignable_v<E>} is \tcode{true}.
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{expected}%
+\begin{itemdecl}
+template<class G>
+  constexpr expected& operator=(const unexpected<G>& e);
+template<class G>
+  constexpr expected& operator=(unexpected<G>&& e);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{GF} be \tcode{const G\&} for the first overload and
+\tcode{G} for the second overload.
+
+\pnum
+\constraints
+\tcode{is_constructible_v<E, GF>} is \tcode{true} and
+\tcode{is_assignable_v<E\&, GF>} is \tcode{true}.
+
+\pnum
+\effects
+\begin{itemize}
+\item
+If \tcode{has_value()} is \tcode{true}, equivalent to:
+\begin{codeblock}
+construct_at(addressof(@\exposid{unex}@), std::forward<GF>(e.value()));
+@\exposid{has_val}@ = false;
+\end{codeblock}
+\item
+Otherwise, equivalent to:
+\tcode{\exposid{unex} = std::forward<GF>(e.value());}
+\end{itemize}
+
+\pnum
+\returns
+\tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarymember{emplace}{expected}%
+\begin{itemdecl}
+constexpr void emplace() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+If \tcode{has_value()} is \tcode{false},
+destroys \exposid{unex} and sets \exposid{has_val} to \tcode{true}.
+\end{itemdescr}
+
+\rSec3[expected.void.swap]{Swap}
+
+\indexlibrarymember{swap}{expected}%
+\begin{itemdecl}
+constexpr void swap(expected& rhs) noexcept(@\seebelow@);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{is_swappable_v<E>} is \tcode{true} and
+\tcode{is_move_constructible_v<E>} is \tcode{true}.
+
+\pnum
+\effects
+See \tref{expected.void.swap}.
+
+\begin{floattable}{\tcode{swap(expected\&)} effects}{expected.void.swap}
+{lx{0.35\hsize}x{0.35\hsize}}
+\topline
+& \chdr{\tcode{this->has_value()}} & \rhdr{\tcode{!has_value()}} \\ \capsep
+\lhdr{\tcode{rhs.has_value()}} &
+  no effects &
+  calls \tcode{rhs.swap(*this)} \\
+\lhdr{\tcode{!rhs.has_value()}} &
+  \seebelow &
+  equivalent to: \tcode{using std::swap; swap(\exposid{unex}, rhs.\exposid{unex});} \\
+\end{floattable}
+
+For the case where \tcode{rhs.value()} is \tcode{false} and
+\tcode{this->has_value()} is \tcode{true}, equivalent to:
+\begin{codeblock}
+construct_at(addressof(@\exposid{unex}@), std::move(rhs.@\exposid{unex}@));
+destroy_at(addressof(rhs.@\exposid{unex}@));
+@\exposid{has_val}@ = false;
+rhs.@\exposid{has_val}@ = true;
+\end{codeblock}
+
+\pnum
+\throws
+Any exception thrown by the expressions in the \effects.
+
+\pnum
+\remarks
+The exception specification is
```

Comment:
If you need to make it a code block due to an overfull `\hbox`, remember to add the colon.
```suggestion
The exception specification is equivalent to
```

Reply:
No. See above.